scriptworker/poll.py

```diff
@@ -121,15 +127,33 @@ def get_azure_urls(context):
 
     Returns:
         dict: the claimTask json
+
+    Raises:
+        ScriptWorkerException: on failure of request_function
     """
+    # this can raise a ScriptWorkerException on failure
     xml = await request_function(context, poll_url)
     for message_info in parse_azure_xml(xml):
-        task = await claim_task(context, message_info['task_info']['taskId'], message_info['task_info']['runId'])
-        if task is not None:
-            log.info("Found task! Deleting from azure...")
-            delete_url = delete_url.replace("{{", "{").replace("}}", "}").format(**message_info)
+        try:
+            task = await retry_async(
+                claim_task,
+                retry_exceptions=(ScriptWorkerRetryException, ),
+                args=(
+                    context, message_info['task_info']['taskId'],
+                    message_info['task_info']['runId']
+                )
+            )
+        except ScriptWorkerRetryException:
+            continue
+        log.debug("Deleting from azure...")
+        delete_url = delete_url.replace("{{", "{").replace("}}", "}").format(**message_info)
+        try:
             response = await request_function(context, delete_url, method='delete', good=[200, 204])
             log.debug(response)
+        except ScriptWorkerException as exc:
+            log.error("Unable to delete from Azure: {}".format(exc))
+            pass
```

Comment:
hm, do we want to raise this? The docs suggest we do.

I'm concerned since we swallow the ScriptWorkerRetryException exception in the retry_async(claim_task) call above and then again with the delete exception here, we may be having instances where we want to possibly sleep at the following line but never do: https://github.com/mozilla-releng/scriptworker/blob/e934a23443a053e1d37ab7c8d6d5a2d14213b465/scriptworker/worker.py#L48

could be parsing wrong though.